apiserver/params/controller.go

```diff
@@ -132,10 +136,138 @@ type ControllerVersionResults struct {
 	GitCommit string `json:"git-commit"`
 }
 
-// DashboardInfo holds the results from an api call
-// to get address info for the juju dashboard.
-type DashboardInfo struct {
-	Addresses []string `json:"addresses"`
-	UseTunnel bool     `json:"use-tunnel"`
-	Error     *Error   `json:"error,omitempty"`
+const (
+	// DashboardConnectionTypeProxy is the type key used for proxy connections
+	DashboardConnectionTypeProxy = "proxy"
+
+	// DashboardConnectionTypeSSHTunnel is the type key used for ssh connections
+	DashboardConnectionTypeSSHTunnel = "ssh-tunnel"
+)
+
+// DashboardConnectionProxy represents a proxy connection to the Juju Dashboard
+type DashboardConnectionProxy struct {
+	Proxier       proxy.Proxier   `json:"proxier"`
+	proxierConfig json.RawMessage `json:"-"`
+	ProxierType   string          `json:"type"`
+}
+
+// ProxierFactory is an interface type representing a factory that can make a
+// new juju proxier from the supplied type and JSON config.
+type ProxierFactory interface {
+	ProxierFromJSONDataBag(proxierType string, config json.RawMessage) (proxy.Proxier, error)
+}
```

Comment:
This doesn't belong here. The params package is for over the wire transfer of data structs. Any logic to act on that data needs to be in the api or apiserver layers.